src/language/class-modifiers.md

```diff
@@ -0,0 +1,240 @@
+A base class can be extended and constructed. This applies trasitively to all
+subtypes, because implementing a subtype implements the supertype as well.  
+
+{{site.alert.note}}
+  Only the `base` modifier can accompany a mixin declaration. 
+  - A `base mixin` can be mixed in, but not implemented.
+  - A `base mixin class` can be constructed, extended, and mixed in, but not
+  implemented.
+{{site.alert.end}}
+
+```dart
+base class Vehicle {
+  String make;
+  String model;
+  void moveForward(int meters) { … }
+}
+
+var myCar = Vehicle(     // Can be constructed
+  make: 'Ford',
+  model: 'T',
+);
+
+class Car extends     // Can be extended
+  Vehicle {
+    int passengers;
+…
+}
+
+class MockVehicle     // ERROR: Cannot be implemented
+  implements Vehicle {
+    @override
+    void moveForward …
+}
+
+```
+
+Disallowing implementation makes `base` useful for:
+- Preventing users of the API from overriding the constructor, for example, to
+bypass its validation.
+- Ensuring all private members are available to subtypes of the base class, and
+and accessing them will never cause a [`noSuchMethod`][] error.
+
+## `interface`
+
+Use the `interface` modifier to **disallow extending a class**.
+An interface class can be implemented and constructed.
+
+```dart
+interface class Vehicle {
+  String make;
+  String model;
+  void moveForward(int meters) { … }
+}
+
+var myCar = Vehicle(     // Can be constructed
+  make: 'Ford',
+  model: 'T',
+);
+
+class Car extends     // ERROR: Cannot be inherited
+  Vehicle {
+    int passengers;
+…
+}
+
+class MockVehicle     // Can be implemented     
+  implements Vehicle {
+    @override
+    void moveForward …
+}
+```
+
+Disallowing inheritance makes `interface` useful for:
+- Defining API interfaces.
+- Preventing users of the API from overriding a subset of methods.
+
+## `final` 
+
+Use the `final` modifier to **disallow extending _and_ implementing** a class.
+Final classes can be constructed.
+
+```dart
+final class Vehicle {
+  String make;
+  String model;
+  void moveForward(int meters) { … }
+}
+
+var myCar = Vehicle(     // Can be constructed
+  make: 'Ford',
+  model: 'T',
+);
+
+class Car extends     // ERROR: Cannot be inherited
+  Vehicle {
+    int passengers;
+…
+}
+
+class MockVehicle     // ERROR: Cannot be implemented     
+  implements Vehicle {
+    @override
+    void moveForward …
+}
+```
+
+Disallowing inheritance and implementation makes `final` useful for:
+- Closing the type hierarchy.
+- Ensuring that adding new members to the class won't risk breaking changes for 
+its consumers.
+- Ensuring access to private members on instances of the final class will never
+throw a [`noSuchMethod`][] error.
+
+## `sealed`
+
+Use the `sealed` modifier to define a class that **cannot be constructed, extended,
+or implementated** outside of the library where the class is defined. 
+
+The purpose of the `sealed` modifier is to enable [_exhaustiveness checking_][]
```

Comment:
I would maybe start with this information -- that the main point of `sealed` is for exhaustiveness. The fact that it can't be subtyped at all outside of its library comes after the exhaustiveness imo.

We prefer users to use `final` if they don't need the exhaustiveness feature.